x/gamm/pool-models/stableswap/amm_test.go

```diff
@@ -702,3 +722,117 @@ func TestCalcSingleAssetJoinShares(t *testing.T) {
 		})
 	}
 }
+
+func TestJoinPoolSharesInternal(t *testing.T) {
```

Comment:
Note: these functions will be more comprehensively tested in https://github.com/osmosis-labs/osmosis/issues/2346 – I only added cases relating to the changes made in this PR so it doesn't get blocked